File: Script/Fattoriale_ricorsivo/Fattoriale.py
import sys

def fattoriale_rec(n: int) -> int:
    if n == 0 or n == 1:
        return 1
    return n * fattoriale_rec(n - 1)

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print(f"Uso: python {sys.argv[0]} <numero>")
        sys.exit(1)

    n = int(sys.argv[1])
    print(f"Fattoriale ricorsivo({n}) = {fattoriale_rec(n)}")

File: Script/Fattoriale_ricorsivo/Fattoriale.js
function fattorialeRec(n) {
  if (n === 0 || n === 1) {
    return 1;
  }
  return n * fattorialeRec(n - 1);
}

if (process.argv.length !== 3) {
  console.log(`Uso: node ${process.argv[1]} <numero>`);
  process.exit(1);
}

const n = parseInt(process.argv[2], 10);
console.log(`Fattoriale ricorsivo(${n}) = ${fattorialeRec(n)}`);
